### DISCUSSION TITLE
How to calculate fluence？

### DISCUSSION THREAD

**Discussion by 1shikang** [nrc-cnrc/EGSnrc#1116] [General]
Dear experts：
  I want to calculate the spatial distribution of fluence. But I only found the FLURZ .The model I want to build in the xyz coordinate system.And I tried writing .egsinp directly, but how to divide the spatial grid instead of the energy grid? I am not concerned about the relationship between flux and energy.Please help me!
Think you!

**Comment by rtownson** [comment]
Using any egs++ application (e.g. egs_chamber or egs_app), you can model any geometry and get fluence scoring using:

https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/classEGS__PlanarFluence.html

Or for volumetric fluence:

https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/classEGS__VolumetricFluence.html

**Reply by 1shikang** [reply]
But how to divide the grid(not energy grid) when using volumetric fluence, because I want to obtain the distribution of fluence in space.

**Reply by ftessier** [reply]
The fluence scoring ausgab object enables you to specify the regions in which to score the fluence with the input:

```
scoring regions = r1 r2 ... rn
```

where `r1 r2 ... rn` is a list of region numbers.

**Reply by 1shikang** [reply]
So if I want to set a lot of counting bins, can I only do it this way?I think this is inconvenient. Alternatively,  Is it  possible to set up  bins in the region directly? Can I obtain the distribution of flux with respect to bin in this way？If i can, please tell me how to do .Think you !

**Reply by rtownson** [reply]
Currently it's necessary to define regions in the geometry that will be your volumetric scoring bins. For example, you could use an XYZ geometry to define voxels, and place that into your geometry (even if the medium is uniform). This allows for a lot more flexibility, because you can define complex grids and non-rectilinear voxels, apply transformations, etc. Note that for the list of regions you can use `region labels` (described on https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/egs_chamber.html), which helps make it easier.

**Reply by 1shikang** [reply]
How should I define the grid? I tried to generate a mesh file in .msh format using the Gmsh software and then import it into EGSnrc, but it showed an error. Is it because I didn't define the materials? However, it seems that Gmsh does not have an option to define materials.
![gmsh导入错误截图](https://github.com/nrc-cnrc/EGSnrc/assets/162414020/98921d0f-ad34-4cb5-a7d1-57e7953dd819)

**Reply by 1shikang** [reply]
I see. This is because I chose a cuboid mesh, and egsnrc only supports importing tetrahedral meshes.. Now I've encountered a new issue related to my physical groups, and I'm not sure where the problem lies. Could you help me with that?
![物理模型问题](https://github.com/nrc-cnrc/EGSnrc/assets/162414020/17586d07-a624-4d29-8f78-666f16a11727)
I think my physical groups name is not a problem.

**Reply by mxxo** [reply]
Without having enough info to confirm, I believe this error is from one of the physical groups in your `msh` file being called `myy`, and then EGSnrc isn't able to find a media with the name `myy` in the media definitions (from your earlier post I assume you're using the `cu521icru.pegs4dat` file to define media).

Are you able to share your `msh` file to confirm this guess? You can also check the `msh` file using a text editor.

**Reply by 1shikang** [reply]
Of course. And I also tried to change the names of the physical groups, but the error still persists. Below are screenshots of my terminal and the Gmsh file.  Thank you.
![错误3](https://github.com/nrc-cnrc/EGSnrc/assets/162414020/8c0d3179-db08-4e14-b90f-61c0bae53955)
[gmsh.zip](https://github.com/nrc-cnrc/EGSnrc/files/15087084/gmsh.zip)

**Reply by mxxo** [reply]
The error message has changed, now it's saying it can't find the `msh` file. Can you try running the same command in the `~/EGSnrc/EGSnrc-master/egs_home/test1` directory (assuming that is where `model3.msh` is saved)?
